### DISCUSSION TITLE
Logitech MX Master 3 for Mac / Forward-Back Buttons not working on MBP M1 2020

### DISCUSSION THREAD

**Discussion by Aldin998** [noah-nuebling/mac-mouse-fix#283] [Questions] [answered]
Hello everyone,

I have and Logitech MX Master 3 for Mac, connected via Bluetooth on my M1 MBP 2020 with Monterey macOS. So I have the following issue, therefore I hope someone can help me fix the problem. I used Logitech Options and Logitech Options + (Beta) but without success. So I found on Reddit the Mac Mouse Fix as an alternative, but the same problem again... 

The left-right clicks work normal, the scroll too but the gesture button, side buttons (forward-backward) and side scroll aren't working. 

Can someone help me fix this issue, or is there anyone with a similar problem? 

PS. I have tried the same mouse on Monterey on an MBP 2016, but the same issue persist, and on my M1 I tried a Xiaomi Bluetooth Mouse, but again side buttons aren't working.

Ty

**Comment by Metagit** [comment]
Hello Aldin,
I also have Issues with the MX Master 3S, meaning the cmd+scroll zoom doesn't work which is super unfortunate...

BUT I might have a solution for at least some of your issues. I had the same issue, that the side buttons for back and forth aren't working. I was able to fix that finally and permanently with "sane side buttons". Ever since I have installed this, I can finally navigate efficiently and comfortably on my Mac. Hope that helps you too.

**Reply by Aldinnn19** [reply]
Hello Metagit, I have fixed my issue, the problem was so dumb -.- I just opened the mouse yesterday and the ribbon cable connecting the side buttons was not connected properly, but I will try the app, I have a mouse that I can try it

**Reply by Metagit** [reply]
Nice, glad to hear that. :) 
Does the cmd+scroll for zoom feature work with your MX Master in safari, finder or picture preview?

**Comment by noah-nuebling** [comment, marked as answer]
Hi guys thanks for reaching out! From what I can tell there were two issues going on for @Aldin998 here. The first is the ribbon cable on the MX Master, and then they got unlucky with the Xiami Mouse because those are known to have hardware issues that prevent the side buttons from working properly when they are connected via bluetooth. See #121.

I'll mark this as answered now to clean things up. If you continue to have problems, please open a new Issue or leave a comment here. Thanks!

**Reply by bpet221** [reply]
That worked for me! my MX master mouse is now being detected in the Mac Mouse Fix 3.0 version. I did some extra steps and am not sure if it played a role. I also deleted Mac Mouse Fix app, deleted MX Master from my bluetooth, and then restarted my computer in addition to your suggestions. 

Now the problem i am having is that I need to use both Logi Options + and MMF...BUT the second i install options + again MMF completley stops working....

Any help here would be amazing! Thanks @noah-nuebling  and team.

**Reply by KaiEtkin** [reply]
did u get this resolved, i also need to use options + and mac mouse fix

**Reply by noah-nuebling** [reply]
Hey guys, using Logitech options alongside MMF is not officially support at the moment. I'll try to fix that at some point, but there's a lot of stuff that I wanna do for the app, and these things usually take me a long time, and so I don't I think this is coming soon. 

My intuition is that, the reason why it sometimes works and sometimes doesn't, is due to the order that MMF and Logitech Options 'tap into' the mouse event stream of macOS. Depending on which app was enabled first, one or the other might be 'tapped into' the mouse event stream before the other one, preventing the other one from handling mouse events properly. 
You can affect the order of the even taps by toggling MMF on and off, or toggling Logitech Options on and off. (Not sure how to toggle Logitech Options on and off - maybe if you kill some Logitech Options helper process from Actitivity Monitor that could do the trick?)

However I realize that always having to toggle the apps on and off for them to work properly is a really bad solution. So I do want to provide a better solution at some point.

In the mean time, I can recommend [Mos](https://mos.caldis.me/), [LinearMouse](https://linearmouse.app/) or [Steer Mouse](https://plentycom.jp/en/steermouse/). Those are the best alternatives to Mac Mouse Fix that I'm aware of. I'm not entirely sure whether LinearMouse and Steer Mouse can coexist with Logitech Options, but I think so.
